pages/_app.js

```diff
@@ -1,53 +1,81 @@
 
-import React from 'react';
-import App from 'next/app';
+import React, { Fragment, useState, useEffect } from 'react';
+import { AnimatePresence } from 'framer-motion';
 
+import { useMemoryStatus } from '../utils/hooks';
 import AnimationEmulationContext from '../components/AnimationEmulationContext';
 
-class MyApp extends App {
-  state = {
-    manualEnabled: false,
-    isAnimationOn: true
-  };
-
-  componentDidMount = () => {
-    const manualEnabled = localStorage.getItem('manual-enabled');
-    const isAnimationOn = localStorage.getItem('is-animation-on');
+const Loading = () => <Fragment>Loading...</Fragment>;
 
+const MyApp = ({ Component, pageProps, router }) => {
+  const [manualEnabled, setManualEnabled] = useState(false);
+  const [isAnimationOn, setIsAnimationOn] = useState(true);
+  useEffect(() => {
     if (manualEnabled) {
-      this.setState({manualEnabled});
+      setManualEnabled(manualEnabled);
     }
 
     if (isAnimationOn) {
-      this.setState({isAnimationOn});
+      setIsAnimationOn(isAnimationOn);
     }
-  };
+  }, []);
 
-  enableManualAnimationHandler = flag => {
-    localStorage.setItem('manual-enabled', flag);
-    this.setState({manualEnabled: flag});
+  const memoryStatus = useMemoryStatus();
+  if (!memoryStatus) return <Loading />;
+  const { overLoaded } = memoryStatus;
+
+  let animationAllowed = true;
+  if (manualEnabled) {
+    animationAllowed = isAnimationOn;
+  } else {
+    animationAllowed = !overLoaded;
+  }
+
+  const enableManualAnimationHandler = flag => {
+    setManualEnabled(flag);
   };
 
-  toggleAnimationHandler = event => {
-    localStorage.setItem('is-animation-on', event.target.checked);
-    this.setState({isAnimationOn: event.target.checked});
+  const toggleAnimationHandler = event => {
+    setIsAnimationOn(event.target.checked);
   };
 
-  render() {
-    const { Component, pageProps, router } = this.props;
-    const { manualEnabled, isAnimationOn } = this.state;
+  if (animationAllowed) {
     return (
```

Comment:
This is very expensive.
```
{ condition ? (
  <AnimatePresence exitBeforeEnter>
    <Component {...pageProps} key={router.route} />
  </AnimatePresence>
) : (
  <Component {...pageProps} key={router.route} />
)}
```

Inside JSX you can insert some logic like above.

Reply:
I have done it.